### DISCUSSION TITLE
React-Joyride is only scrolling the closest parent scroll container

### DISCUSSION THREAD

**Discussion by jiyang-sas** [gilbarbara/react-joyride#833] [General]
React-Joyride is only scrolling the closest parent scroll container and does not actually scroll the whole page, it seems that it does not really support the nested scroll container case.
Here is a codesandbox snipet:

https://codesandbox.io/s/joyride-with-nested-scroll-container-x0d8pi?file=/src/App.tsx

In the above example, if you use the outer scrollbar to hide the text 'Content is here', then start the tour, you will see that  it scrolls up the inner container while leaving the outer container scrolled to the bottom of it's scroll container.

Any idea on how to fix this? Thank you!

**Comment by jiyang-sas** [comment]
@gilbarbara Is there any joyride callbacks  we can use to check  if the highlighted element is in view? thanks!

**Reply by gilbarbara** [reply]
Did you check the [docs](https://docs.react-joyride.com/props)? There's only one `callback` prop that is called multiple times when the internal state changes.

**Reply by jiyang-sas** [reply]
Thanks!
